rskj-core/src/main/java/co/rsk/net/NodeMessageHandler.java

```diff
@@ -66,7 +63,7 @@ public class NodeMessageHandler implements MessageHandler, InternalService, Runn
     private final PeerScoringManager peerScoringManager;
 
     private final StatusResolver statusResolver;
-    private final Set<Keccak256> receivedMessages = Collections.synchronizedSet(new HashSet<>());
+    private final Map<Peer, Set<Keccak256>> receivedPeerMessages = Collections.synchronizedMap(new HashMap<>());
```

Comment:
I think ConcurrentHashMap could be used instead.

Reply:
I don't know the context why `synchronizedSet` was selected in the past, I just followed the pattern. I just assumed they had a valid reason.

Maybe @Vovchyk can give more context here?